src/lib/provable/merkle-map.ts

```diff
@@ -13,15 +13,25 @@ class MerkleMap {
   /**
    * Creates a new, empty Merkle Map.
    * @returns A new MerkleMap
+   * @example
+   * ```ts
+   * const merkleMap = new MerkleMap();
+   * ```
    */
   constructor() {
     this.tree = new MerkleTree(256);
   }
 
   _keyToIndex(key: Field) {
     // the bit map is reversed to make reconstructing the key during proving more convenient
-    let bits = BinableFp.toBits(key.toBigInt()).reverse();
+    let bits = BinableFp.toBits(key.toBigInt());
 
+    // Make sure that the key fits in 254 bits, in order to avoid collisions since the Pasta field modulus is smaller than 2^255
+    if (bits.length > 254) {
```

Comment:
I think it will always be length 255, but you can check if the last entry is 0

Reply:
oops! Good catch!